File: mlir/examples/dsp/SimpleBlocks/Output/TryDSPApps/BenchmarkTest/DSP-DSL/hearingAid.py
def main() {
var fs = 8000;
  # var step = 1/8000; 
  # print(step);
	var input = getRangeOfVector(0, 100000000, 0.000125);
  var f_sig = 500;
  var pi = 3.14159265359;
  var getMultiplier = 2 * pi * f_sig;
  # print(getMultiplier);
  var getSinDuration = gain(input, getMultiplier);
  # print(getSinDuration);
  var clean_sig = sin(getSinDuration );

  #define a noise signal with freq = 3000
  var f_noise = 3000;
  var getNoiseSinDuration = gain(input, 2 * pi * f_noise);
  var noise = sin(getNoiseSinDuration);
  var noise1 = gain(noise, 0.5);

  var noisy_sig = clean_sig + noise1;
  # print(noisy_sig);
  # print(clean_sig);
  var mu = 0.01;
  var filterSize = 32;
  var y = lmsFilterResponse(noisy_sig, clean_sig, mu, filterSize);
  var G1 = 1002300;
  var sol = gain(y,G1);
  # print(y);
  print(sol);

}

File: mlir/examples/dsp/SimpleBlocks/Output/TryDSPApps/BenchmarkTest/Matlab/hearingAid.m
% Define constants
INPUT_LENGTH = 100000000;
fs = 8000;
step = 1 / fs;

% Generate input range
input = (0:step:(INPUT_LENGTH-1)*step)';

% Generate clean signal
f_sig = 500;
clean_sig = sin(2 * pi * f_sig * input);

% Generate noise signal with frequency of 3000 Hz
f_noise = 3000;
noise = 0.5 * sin(2 * pi * f_noise * input);

% Create noisy signal by adding noise to clean signal
noisy_sig = clean_sig + noise;

% LMS filter response function
function y = lmsFilterResponse(noisy_sig, clean_sig, mu, filterSize)
    w = zeros(filterSize, 1);
    y = zeros(size(noisy_sig));
    
    for n = 1:length(noisy_sig)
        x = noisy_sig(max(1, n-filterSize+1):n);
        x = [zeros(filterSize - length(x), 1); x];
        y(n) = w' * x;
        e = clean_sig(n) - y(n);
        w = w + mu * e * x;
        y(n) = e;
    end
end

% Apply LMS filter
mu = 0.01;
filterSize = 32;
y = lmsFilterResponse(noisy_sig, clean_sig, mu, filterSize);

% Apply final gain factor G1 to the LMS filter output
G1 = 1002300;
sol = G1 * y;

% Display 
disp(sol);